src/Edge.ts

```diff
@@ -521,15 +538,6 @@ export class Edge {
         if (type === "warning" && errorsWarnings !== null){
             errorsWarnings.warnings.push(issue);
         }
-    }
-}
-
-export namespace Edge {
-    export enum Validity {
-        Unknown = "Unknown",        // validity of the edge is unknown
-        Impossible = "Impossible",  // never useful or valid
-        Invalid = "Invalid",        // invalid, but possibly useful for expert users?
-        Warning = "Warning",        // valid, but some issue that the user should be aware of
-        Valid = "Valid"             // fine
+        Eagle.getInstance().logicalGraph().findEdgeById(edgeId)?.issues().push({issue:issue, validity:linkValid})
```

Comment:
Could we avoid this lookup of the edge from its id? By passing the edge (instead of edgeId) into isValidLog()